provider/core/src/data_provider.rs

```diff
@@ -797,6 +801,29 @@ where
     }
 }
 
+impl DataPayload<BufferMarker> {
+    /// Converts a reference-counted byte buffer into a `DataPayload<BufferMarker>`.
+    pub fn from_rc_buffer(buffer: Rc<[u8]>) -> Self {
+        Self {
+            yoke: Yoke::attach_to_rc_cart(buffer).wrap_cart_in_option(),
```

Comment:
suggestion (nb): `Self::from_yoked_buffer(Yoke::attach_to_rc_cart(buffer))`?

Reply:
I prefer writing out the full Yoke fluent chain each time.

Reply:
wfm!